aiida_vasp/calcs/immigrant.py

```diff
@@ -0,0 +1,115 @@
+# pylint: disable=abstract-method
+# explanation: pylint wrongly complains about (aiida) Node not implementing query
+"""VaspImmigrant calculation: Immigrate externally run VASP calculations into AiiDA."""
+from py import path as py_path  # pylint: disable=no-name-in-module,no-member
+from aiida.common.exceptions import InputValidationError
+from aiida.common.lang import override
+from aiida.work import JobProcess
+
+from aiida_vasp.data.potcar import PotcarData
+from aiida_vasp.io.incar import IncarParser
+from aiida_vasp.io.kpoints import KpParser
+from aiida_vasp.io.poscar import PoscarParser
+from aiida_vasp.io.potcar import MultiPotcarIo
+from aiida_vasp.io.chgcar import ChgcarParser
+from aiida_vasp.io.wavecar import WavecarParser
+from aiida_vasp.utils.aiida_utils import get_data_node
+
+
+def get_quantity_node(parser, quantity):
+    return parser.get_quantity(quantity, None)[quantity]
+
+
+class VaspImmigrantJobProcess(JobProcess):
+    """
+    JobProcess subclass for importing non-aiida VASP runs.
+
+    Simulate running the VaspCalculation up to the point where it can be retrieved and parsed,
+    then hand over control to the runner for the rest.
+    """
+
+    @override
+    def run(self):
+        import plumpy
+        from aiida.common.datastructures import calc_states
+        from aiida.common.links import LinkType
+        from aiida.work.job_processes import RETRIEVE_COMMAND
+        from aiida.orm.calculation.job import _input_subfolder
+
+        _ = super(VaspImmigrantJobProcess, self).run()
+
+        state = self.calc.get_state()
+
+        if state != calc_states.TOSUBMIT:
```

Comment:
When can this happen?

Reply:
This is too defensive and should go, I struggle to think of a reasonable use case where this could happen, only in development but it would be noticeable and the check could be added then with a more concrete error message.